pkg/plugins/handlers.go

```diff
@@ -17,15 +17,42 @@ import (
 type PluginsHandler struct {
 	Client             *http.Client
 	PluginsEndpointMap map[string]string
+	PublicDir          string
 }
 
-func NewPluginsHandler(client *http.Client, token string, pluginsEndpointMap map[string]string) *PluginsHandler {
+func NewPluginsHandler(client *http.Client, pluginsEndpointMap map[string]string, publicDir string) *PluginsHandler {
 	return &PluginsHandler{
 		Client:             client,
 		PluginsEndpointMap: pluginsEndpointMap,
+		PublicDir:          publicDir,
 	}
 }
 
+func (p *PluginsHandler) HandleLocales(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "GET" {
+		w.Header().Set("Allow", "GET")
+		serverutils.SendResponse(w, http.StatusMethodNotAllowed, serverutils.ApiError{Err: "Method unsupported, the only supported methods is GET"})
+		return
+	}
+
+	query := r.URL.Query()
+	lang := query.Get("lng")
+	// In case of static plugin the localization namespace should contain name of the json file that contains the localization strings,
+	// since various static plugins have different names for the localization file.
+	// eg. namespace query for the local-storage-operator-plugin locales - '&ns=lso-plugin.json'
+	// In case of the dynamic plugin the localization namespace should contain name of the plugin prefixed with 'plugin:' prefix
+	// No file extension is needed since for the dynamic plugins we will fetch 'default.json' from the plugin pod.
+	// eg. 'plugin:helm-plugin'
```

Comment:
Can you open a PR to update the dynamic plugin enhancement with this proposal? Then we can get agreement from stakeholders.

cc @christianvogt @vojtechszocs

Reply:
@spadgett https://github.com/openshift/enhancements/pull/815